clients/unshaded/src/main/java/tachyon/client/block/LocalBlockOutStream.java

```diff
@@ -116,13 +113,11 @@ public void flush() throws IOException {
     if (mReservedBytes < bytesToWrite) {
       mReservedBytes += requestSpace(bytesToWrite - mReservedBytes);
     }
-    MappedByteBuffer mappedBuffer =
-        mLocalFileChannel.map(FileChannel.MapMode.READ_WRITE, mFlushedBytes, bytesToWrite);
-    mappedBuffer.put(mBuffer.array(), 0, bytesToWrite);
-    BufferUtils.cleanDirectBuffer(mappedBuffer);
+    mBuffer.flip();
+    mWriter.append(mBuffer);
```

Comment:
calling `flush()` twice in a row without writing should have `bytesToWrite = 0`, right?